### PR TITLE
[Fixes #9510] Fix timestamp placement for text elements

### LINKED ISSUE #9510: Timestamps in message bubble always displayed on second line
# Bug Report

## Problem

Timestamp in message bubbles always displayed on next line after message content. We need to include width of timestamp value and if it fits (along with the text) message bubble max width -> to display it on the same line.

#### Expected behavior

If timestamp fits message bubble max width - it  is displayed on the same line with text
Screenshot 2019-11-22 at 13 54 48

#### Actual behavior

Timestamps in message bubble always occupy the next line

Screenshot 2019-11-22 at 13 54 55

### Reproduction

- Open Status and create new account
- Join public chat and send message with 3 chars

### Additional Information
- Status version: develop 0.14.0 (2019112202)
- Operating System: Android and iOS

### PR DESCRIPTION
**Pull request by cammellos**
status: ready

### CODE REVIEW

File: src/status_im/ui/screens/chat/message/message.cljs [hunk 1]
```diff
@@ -139,9 +139,15 @@
...
 
                           tree]
-  (conj (reduce (fn [acc e] (render-block message acc e)) [react/view {}] tree)
-        [react/text {:style (style/message-timestamp-placeholder outgoing)}
-         (str "  " timestamp-str)]))
+  (let [elements (reduce (fn [acc e] (render-block message acc e)) [react/view {}] tree)
+        timestamp [react/text {:style (style/message-timestamp-placeholder outgoing)}
+                   (str "  " timestamp-str)]
+        last-element (peek elements)]
+    (if (= react/text-class (nth last-element 0))
```
Comment (vitvly) [inline src/status_im/ui/screens/chat/message/message.cljs:149]: Can we use `first` instead of `nth` here?

**Review by vitvly — APPROVED** [review]

File: src/status_im/ui/screens/chat/message/message.cljs (hunk 1 above)
Comment (cammellos) [reply src/status_im/ui/screens/chat/message/message.cljs:149]: I benchmarked `first` vs `nth` and roughly `nth` is 30% faster, and being a critical path I would be inclined to keep `nth` (although probably at some point we might go for `aget` and pure javascript for this structures, as it's much much faster, roughly 90% faster then first), but don't mind either ways

Comment (vitvly) [reply src/status_im/ui/screens/chat/message/message.cljs:149]: Oh wow, thanks for important information! Can you then please include a short comment with the performance rationale?

Comment (cammellos) [reply src/status_im/ui/screens/chat/message/message.cljs:149]: sure!

**Review by flexsurfer — APPROVED** [review]

**Review by yenda — APPROVED** [review]

**Comment by churik** [comment]
@cammellos in failed e2e nothing related to your PR

**Pull request merged**